src/main.rs

```diff
@@ -16,264 +16,178 @@ use std::{
     collections::hash_map::DefaultHasher,
     env,
     hash::{Hash, Hasher},
+    ops::Deref,
     string::ToString,
 };
 
+#[derive(Copy, Clone, PartialEq, Eq, PartialOrd, Ord, ValueEnum)]
 enum PlanFormat {
     Json,
     Toml,
 }
 
-impl PlanFormat {
-    fn from_str(s: &str) -> Result<Self> {
-        match s {
-            "json" => Ok(PlanFormat::Json),
-            "toml" => Ok(PlanFormat::Toml),
-            _ => bail!("Invalid plan format"),
-        }
-    }
+#[derive(Parser)]
+#[command(author, version, about, long_about = None)]
+#[command(propagate_version = true)]
+struct Args {
+    #[command(subcommand)]
+    command: Commands,
+
+    /// Specify an entire build plan in json format that should be used to configure the build
+    #[arg(long, global = true)]
+    json_plan: Option<String>,
+
+    /// Specify the install command to use
+    #[arg(long, short, global = true)]
+    install_cmd: Option<String>,
+
+    /// Specify the build command to use
+    #[arg(long, short, global = true)]
+    build_cmd: Option<String>,
+
+    /// Specify the start command to use
+    #[arg(long, short, global = true)]
+    start_cmd: Option<String>,
+
+    /// Provide additional nix packages to install in the environment
+    #[arg(long, short, global = true)]
+    pkgs: Vec<String>,
+
+    /// Provide additional apt packages to install in the environment
+    #[arg(long, short, global = true)]
+    apt: Vec<String>,
+
+    /// Provide additional nix libraries to install in the environment
+    #[arg(long, short, global = true)]
+    libs: Vec<String>,
```

Comment:
This can't have a short option since it conflicts with labels

![image](https://user-images.githubusercontent.com/3044853/202238771-0f6efc36-047a-4285-b558-5733483e68ed.png)